src/components/Dialog/DialogsManager.ts

```diff
@@ -0,0 +1,133 @@
+import { StateStore } from 'stream-chat';
+
+type DialogId = string;
+
+export type GetOrCreateParams = {
+  id: DialogId;
+};
+
+export type Dialog = {
+  close: () => void;
+  id: DialogId;
+  isOpen: boolean | undefined;
+  open: (zIndex?: number) => void;
+  remove: () => void;
+  toggle: () => void;
+  toggleSingle: () => void;
+};
+
+type DialogInitOptions = {
+  id?: string;
+};
+
+type Dialogs = Record<DialogId, Dialog>;
+
+type DialogsManagerState = {
+  dialogs: Dialogs;
+  openDialogCount: number;
+};
+
```

Comment:
I feel like instead of keeping (and managing) an actual value of opened dialogs, you can create a reusable selector (even hook utilising this selector) which derives this value from the actual state of the dialogs. This is less error-prone and increases reliability in terms of future updates to the manager itself (less things to care about);

```ts
const openedDialogCountSelector = (nextValue: DialogManagerState) => {
    return Object.values(nextValue.dialogsById).reduce((count, dialog) => {
        if (dialog.isOpen) return count + 1;
        return count;
    }, 0)
}
```
and use like so:

```ts
// manager comes from parameter for example purposes
const useOpenedDialogCount = (manager) => {
    return useStateStore(manager.state, openedDialogCountSelector);
}
```

Reply:
Done